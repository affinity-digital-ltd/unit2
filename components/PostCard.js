import { Component } from 'react'
import moment from 'moment'
import Link from 'next/link'

class PostCard extends Component {
  truncate (string) {
    if (string.length > 99) {
      return `${string.substr(0, 99)} ...`
    } else {
      return string
    }
  }

  render () {
    let {post} = this.props

    return <div className='col-xs-12 col-md-6 col-lg-3' style={{alignItems: 'stretch', display: 'flex'}}>
      <div className='card c-card'>
        <div className='card-body'>
          <h5 className='card-title'><Link href='post_path(post)'><a>{post.title}</a></Link></h5>
          <h6 className='card-subtitle mb-2 text-muted'><time dateTime={post.created_at}>{moment(post.created_at).format('LL')}</time></h6>
          <p className='card-text'>{this.truncate(post.intro)}</p>
        </div>
      </div>
    </div>
  }
}

export default PostCard
