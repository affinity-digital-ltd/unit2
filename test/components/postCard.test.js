import PostCard from '../../components/PostCard'

describe('<PostCard />', () => {
  test('it renders correctly', () => {
    const post = {
      id: '1',
      title: 'This is my article',
      intro: 'This is the intro text to my article',
      created_at: new Date(Date.UTC(96, 1, 2, 3, 4, 5))
    }
    const component = shallow(
      <PostCard post={post} />
    )

    expect(component).toMatchSnapshot()
  })
})
