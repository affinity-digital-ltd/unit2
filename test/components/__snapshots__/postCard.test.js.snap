// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> it renders correctly 1`] = `
<div
  className="col-xs-12 col-md-6 col-lg-3"
  style={
    Object {
      "alignItems": "stretch",
      "display": "flex",
    }
  }
>
  <div
    className="card c-card"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        <Link
          href="/post?id=1"
        >
          <a>
            This is my article
          </a>
        </Link>
      </h5>
      <h6
        className="card-subtitle mb-2 text-muted"
      >
        <time
          dateTime={1996-02-02T03:04:05.000Z}
        >
          February 2, 1996
        </time>
      </h6>
      <p
        className="card-text"
      >
        This is the intro text to my article
      </p>
    </div>
  </div>
</div>
`;
